rails-commenting.rb

```diff
@@ -7,24 +7,30 @@
 # FILE: app/controller/blog_posts_controller.rb
 
 # ---1)
+#The controller file contains code for the Class, it's a child class of Application controller, since it's a type of controller. After the route decides which controller and method to run from the url, the controller contains all the methods that run the CRUD actions that will be available to the view or API. The first method is being defined here as index, which represents the 'R' "read" action in CRUD.
 class BlogPostsController < ApplicationController
   def index
     # ---2)
+#The instance variable "posts" is being assigned a value that comes from the function being executed here. Here, the function gets all the objects of the BlogPost table through Active Record. Then in the view file, the developer can access the value by using that instance variable. Normally I'd take the list of all the BlogPosts to be rendered in the view with this method.
     @posts = BlogPost.all
   end
 
   # ---3)
+#The method "show" is also a read action, which shows a specific post. This method accesses a specific row in the database table (BlogPost) according to the :id parameter set from the route. The route checks the value for that parameter from the url and the controller passes it through the method. In this case it picks out the particular blog post by the primary key.
   def show
     @post = BlogPost.find(params[:id])
   end
 
   # ---4)
+# The new method represents a midway point between Create and Read from CRUD. The .new method on BlogPost will create a new BlogPost object but will not save it in the database, unless you run @post.save. This can be used alternatively to the create method if you'd like more tasks in the function before the object is added to the database.
```

Comment:
What is a unique item to the new method? It is something that the user will see.